test/silhouette.jl

```diff
@@ -34,4 +34,18 @@ c = [3, 1]
 
 @test silhouettes(a, c, D) ≈ [0.5, 0.5, -1/3, 0.0]
 
+a = [fill(1, 5); fill(2, 5)]
```

Comment:
It's not yet widely used in Clustering.jl, but I think it's a good practice to put the tests inside `@testset "silhouettes() handle zero distances correctly" begin ... end` block.